### PR TITLE
Adding SAI Challenger with DASH enhancements and test cases

### PR DESCRIPTION
**Pull request by chrispsommers**
1. Introduced a new way to develop tests - DASH-like test configuration used for both DUT and traffic generator.
2. Added scalable test examples for VNET2VNET use case with SAI configuration generation using “saigen” module.
3. Added OTG/snappi helper modules to scale traffic profile.
4. Added SAI-Challenger submodule and testbed definitions (definition only, testbed itself stays without changes)
5. Extended existing dash-saithrift-client with SAI-Challenger client modules and confgen requirements.
6. Enabled new test cases on CI
7. Documentation

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -2,3 +2,10 @@
 	path = dash-pipeline/SAI/SAI
 	url = https://github.com/reshmaintel/SAI.git
 	branch = dash-ptf-ci
+[submodule "test/SAI-Challenger"]
+	path = test/SAI-Challenger
+	url = https://github.com/opencomputeproject/SAI-Challenger
```
Comment (lguohan) [inline .gitmodules:7]: not sure we can support sai challenger for dash project.

**Review by lguohan — CHANGES_REQUESTED** [review]
need to have discussion on introducing sai challenger to the test repo.

**Comment by chrispsommers** [comment]
> need to have discussion on introducing sai challenger to the test repo.

Plenty of time to discuss, this is a draft PR and the presentation to the community is to be resumed shortly. Thanks.

File: .gitmodules [hunk 2]
```diff
@@ -2,3 +2,10 @@
...
+	branch = multiple-api-support
+[submodule "dash-pipeline/cgyang"]
+	path = dash-pipeline/cgyang
+	url = https://github.com/mgheorghe/cgyang
```
Comment (lguohan) [inline .gitmodules:11]: why personal repo?

**Comment by chrispsommers** [comment]
Replaced by https://github.com/Azure/DASH/pull/278

**Pull request closed without merging**